### PR TITLE
(MAINT) Remove testing logic for Puppet 4.5

### PR DESCRIPTION
**Pull request by b4ldr**
Support for puppet 4.5 was dropped from stdlib some time ago however the 
spec tests all include an if guard to prevent them running on puppet
4.5.  As this is no longer required this PR cleans up theses guards

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
Looks like it includes https://github.com/puppetlabs/puppetlabs-stdlib/pull/1192 now. I'd prefer to keep that separate. Also looks like Rubocop is unhappy.

**Comment by b4ldr** [comment]
> Looks like it includes #1192 now. I'd prefer to keep that separate. Also looks like Rubocop is unhappy.

good catch that was over-site should be removed now

File: spec/type_aliases/ip_address_v6_nosubnet_full_spec.rb [hunk 1]
```diff
@@ -1,28 +1,24 @@
 require 'spec_helper'
 
-if Puppet::Util::Package.versioncmp(Puppet.version, '4.5.0') >= 0
-  describe 'Stdlib::IP::Address::V6::Nosubnet::Full' do
```
Comment (ekohl) [inline spec/type_aliases/ip_address_v6_nosubnet_full_spec.rb:4]: I think this line should not have been removed

File: types/httpstatus.pp [hunk 2]
```diff
@@ -1 +1 @@
-type Stdlib::HttpStatus = Integer[100, 599]
+type Stdlib::HttpStatus = Stdlib::Http::Status
```
Comment (ekohl) [inline types/httpstatus.pp:1]: This is also unrelated

**Review by ekohl — CHANGES_REQUESTED** [review]

File: spec/type_aliases/ip_address_v6_nosubnet_full_spec.rb (hunk 1 above)
Comment (b4ldr) [reply spec/type_aliases/ip_address_v6_nosubnet_full_spec.rb:4]: added back

**Review by ekohl — APPROVED** [review]
I'll let others take a look as well.
